### ISSUE TITLE
Grad CAM computation

### ISSUE THREAD

**Issue by kapytaine** [openitnovem/readml#12]
I think there is a not required epsilon in the computation of the grad CAM in the current code. In the file grad_cam.py, here are the two lines that use the argument `eps`.
```
# use automatic differentiation to compute the gradients
grads = tape.gradient(loss, conv_outputs) + eps
```

```
# normalize the heatmap such that all values lie in the range [0, 1], scale the resulting
# values, and then convert to an unsigned 8-bit integer
heatmap = (heatmap - np.min(heatmap)) / ((heatmap.max() - heatmap.min()) + eps)
```

I do understand the purpose of the second eps in order "to avoid ending with a denominator of 0" as written in the docstring. However, I don't understand the purpose of the first one in the gradient computation. I would suggest to remove it if it is not necessary.

**Comment by JonTri-5** [comment]
I have tried to retire the ```eps``` value. But, it create me an error, however, if you want to retire this epsilon, this possible to pass ```persistent```argument to ```True```into ```tf.GradientTape```.

```python
with tf.GradientTape(persistent=True) as tape:
   ...
```

But, thanks to the tests, I have seen that grad cam computation results are impacted. Many results are dark without heatmap. We don't have a correct intelligibility with grad cam for all picture if we retire the ```eps``` value.

Finally, this is better to unchange the code. That's why I don't have commit anything.

**Comment by kapytaine** [comment]
I tried to remove the first epsilon. The test dl_explain_image_grayscale still works.

The grad cam results are impacted because of the relative high value of `eps` compared to `heatmap.max() - heatmap.min()` in te following line `heatmap = (heatmap - np.min(heatmap)) / ((heatmap.max() - heatmap.min()) + eps)`. I tried to remove the first epsilon and reduce the default value of epsilon. The results seem as expected.

Consequently I would suggest to remove the first epsilon and handle the second issue. I don't know exactly what to do with this second aspect (maybe decrease the default value of epsilon, log a warning when eps and heatmap values are similar and heatmap almost constant, rescale heatmap between 0 and 1 after using the epsilon).
